File: 6.Js版每日一题/509. 斐波那契数.js
// 509. 斐波那契数
var fib = function (N) {
	return N === 0 ? 0 : N < 3 ? 1 : fib(N - 1) + fib(N - 2)
}

// 动态规划
var fib = function (N) {
	let dp = new Uint32Array(N + 1)
	let i = 1
	dp[1] = 1
	while (i++ < N) dp[i] = dp[i - 1] + dp[i - 2]
	return N ? dp[N] : 0
}

File: 7.Go版每日一题/509. 斐波那契数.go
package main

// 509. 斐波那契数
func fib(n int) int {
	if n < 2 {
		return n
	}
	//  res , [p, q]
	p, q, res := 0, 0, 1
	for i := 2; i <= n; i++ {
		p = q
		q = res
		res = p + q
	}
	return res
}
func main509() {

}
